pyproject.toml

```
@@ -201,3 +207,9 @@ known-first-party = ["reigns", "towns", "lands", "items", "users"]
 [tool.ruff.lint.pyupgrade]
 # Preserve types, even if a file imports `from __future__ import annotations`.
 keep-runtime-typing = true
+
+[tool.ruff.lint.flake8-tidy-imports]
+banned-module-level-imports = ["typer.rich_utils"]
```

This might just be me being confused, but shouldn't this be 
```suggestion
banned-module-level-imports = ["rich"]
```
?

---

This config is specifically to force imports of `typer.rich_utils` to stay within functions, i.e. they shouldn't be done at the top of the module at the module level (hence "`banned-module-level-imports`").

The `banned-api` config below is the one to prevent directly importing `rich` anywhere (but we have a per-file exception for `rich_utils`).

I will add a comment to clarify.

---

Ah, thanks for clarifying. I didn't quite get that L215-L216 both define the blocked API ànd define the error msg - I thought it did only the latter. Clear, tnx!